Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
I understand your point but it's how work the PHP standard library and doing another way leads to unexpected behaviors when dealing with large numbers (especially in this case): http://php.net/manual/en/function.ceil.php